openslides_backend/action/actions/motion/create_forwarded.py

```diff
@@ -215,42 +40,8 @@ def check_permissions(self, instance: dict[str, Any]) -> None:
             msg = "Amendments cannot be forwarded."
             raise PermissionDenied(msg)
 
-    def set_origin_ids(self, instance: dict[str, Any]) -> None:
-        if instance.get("origin_id"):
-            origin = self.datastore.get(
-                fqid_from_collection_and_id("motion", instance["origin_id"]),
-                ["all_origin_ids", "meeting_id"],
-            )
-            instance["origin_meeting_id"] = origin["meeting_id"]
-            instance["all_origin_ids"] = origin.get("all_origin_ids", [])
-            instance["all_origin_ids"].append(instance["origin_id"])
-
-    def check_state_allow_forwarding(self, instance: dict[str, Any]) -> None:
-        origin = self.datastore.get(
-            fqid_from_collection_and_id(self.model.collection, instance["origin_id"]),
-            ["state_id"],
-        )
-        state = self.datastore.get(
-            fqid_from_collection_and_id("motion_state", origin["state_id"]),
-            ["allow_motion_forwarding"],
-        )
-        if not state.get("allow_motion_forwarding"):
-            raise ActionException("State doesn't allow to forward motion.")
+    def create_amendments(self, amendment_data: ActionData) -> ActionResults | None:
+        return self.execute_other_action(MotionCreateForwardedAmendment, amendment_data)
 
-    def get_history_information(self) -> HistoryInformation | None:
-        forwarded_entries = defaultdict(list)
-        for instance in self.instances:
-            forwarded_entries[
-                fqid_from_collection_and_id("motion", instance["origin_id"])
-            ].extend(
-                [
-                    "Forwarded to {}",
-                    fqid_from_collection_and_id("meeting", instance["meeting_id"]),
-                ]
-            )
-        return forwarded_entries | {
-            fqid_from_collection_and_id("motion", instance["id"]): [
-                "Motion created (forwarded)"
-            ]
-            for instance in self.instances
-        }
+    def should_forward_amendments(self, instance: dict[str, Any]) -> bool:
+        return instance.pop("with_amendments", False)
```

Comment:
Dangerous combination of method name (seems to be idempotent like a "variable") and internal behavior (definitely not idempotent). Please change behavior or name (=> pop_with_amendment)

Reply:
Changed the function to fit the name instead